src/Dofs/ConstraintHandler.jl

```diff
@@ -1762,3 +1821,146 @@ function _condense_local!(
+    user_provided > 0 && return user_provided
+    return _default_bc_qr_order(ip)
+end
+# Q&D default, should be more elaborated
+_default_bc_qr_order(::Interpolation{<:Any, order}) where {order} = 2 * order
+
+function add!(ch::ConstraintHandler, dbc::WeakDirichlet)
+    # Duplicate the Dirichlet constraint for every SubDofHandler
+    dbc_added = false
+    for sdh in ch.dh.subdofhandlers
+        # Skip if the constrained field does not live on this sub domain
+        dbc.field_name in sdh.field_names || continue
+        # Compute the intersection between dbc.set and the cellset of this
+        # SubDofHandler and skip if the set is empty
+        filtered_set = filter_dbc_set(get_grid(ch.dh), sdh.cellset, dbc.facets)
+        isempty(filtered_set) && continue
+        # Fetch information about the field on this SubDofHandler
+        field_idx = find_field(sdh, dbc.field_name)
+        interpolation = getfieldinterpolation(sdh, field_idx)
+        CT = getcelltype(sdh) # Same celltype enforced in SubDofHandler constructor
+        qr_order = _default_bc_qr_order(dbc.qr_order, interpolation)
+        fqr = FacetQuadratureRule{getrefshape(interpolation)}(qr_order)
+        fv = FacetValues(fqr, interpolation, geometric_interpolation(CT))
+        local_facet_dofs, local_facet_dofs_offset =
+            _local_facet_dofs_for_bc(get_base_interpolation(interpolation), 1, 1, field_offset(sdh, field_idx), dirichlet_facetdof_indices)
+        facet_dofs = ArrayOfVectorViews(local_facet_dofs_offset, local_facet_dofs, LinearIndices(1:(length(local_facet_dofs_offset) - 1)))
+
+        filtered_dbc = WeakDirichlet(dbc.f, filtered_set, dbc.field_name, qr_order, fv, facet_dofs)
+
+        _add!(ch, filtered_dbc, facet_dofs)
+
+        dbc_added = true
+    end
+    dbc_added || error("No overlap between dbc::Dirichlet and fields in the ConstraintHandler's DofHandler")
+    return ch
+end
+
+function _add!(ch::ConstraintHandler, dbc::WeakDirichlet, facet_dofs)
+    # loop over all the faces in the set and add the global dofs to `constrained_dofs`
+    constrained_dofs = Int[]
+    cc = CellCache(ch.dh, UpdateFlags(; nodes = false, coords = false, dofs = true))
+    for (cellidx, facetidx) in dbc.facets
+        reinit!(cc, cellidx)
+        local_dofs = facet_dofs[facetidx]
+        for d in local_dofs
+            push!(constrained_dofs, cc.dofs[d])
+        end
+    end
+
+    # save it to the ConstraintHandler
+    push!(ch.idbcs, dbc)
+    for d in constrained_dofs
+        add_prescribed_dof!(ch, d, NaN, nothing)
+    end
+    return ch
+end
+
+
+function _update_weak_dbc!(
+        inhomogeneities::Vector{T}, f::Function, facets::AbstractVecOrSet{FacetIndex}, fv::FacetValues, facet_dofs::ArrayOfVectorViews,
+        dh::AbstractDofHandler, dofmapping::Dict{Int, Int}, dofcoefficients::Vector{Union{Nothing, DofCoefficients{T}}}, time::Real
+    ) where {T}
+    ip = function_interpolation(fv)
+    n = maximum(length, dirichlet_facetdof_indices(ip))
```

Comment:
Is `n` the number of dofs related to a single facet? So for low orders `n = 1` and we solve a scalar eq?

Reply:
Yes. 
In general, we could do a lot of performance optimization here if required, but when I did some benchmark I didn't see that as a bottleneck in the Maxwell tutorial.